### ISSUE TITLE
Some kind of protection

### ISSUE THREAD

**Issue by GoogleCodeExporter** [IvanVolosyuk/wifikeyboard#17]
```
Using it on Nexus One, froyo 2.2 - i'd very apreciate this app, but I'd like to 
point out that portscanning a network for port "7777" would make possible for 
virtually anyone to operate your android smartphone.

Would it be possible to add a password request (or at least custom port 
number)? Both will be nice.

Thank you very much.

```

Original issue reported on code.google.com by `qualsiasi` on 25 Aug 2010 at 6:29

**Comment by mofosyne** [comment]
To add to this, what I found useful about the security of "signals" a secure instant messenger/VOIP, was the passphrase. During a VOIP call both phone during an active secure session would display a common pass phrase after a successful key exchange. So if you kept in a keyboard stub like in Remote Keyboard app, you could display a passphrase that must match the passphrase on the remote keyboard webpage.

So assuming that the javascript in your html page you received is not compromised, a matching passphrase indicates that you can trust what you type is secure.

* https://github.com/onyxbits/remotekeyboard

**Comment by IvanVolosyuk** [comment]
JavaScript _can_ be compromised.
The secure way would be ssh port forwarding. From my side it should be an
option to listen on local connections only.

**Comment by mofosyne** [comment]
Good point. I do find setting up ssh to be more involved if you never regularly use the terminal. If possible I would certainly prefer a "https" option, but otherwise I wouldn't mind if there is a tutorial on the app on how to setup ssh if you don't know what you are doing (e.g. A help button). (Perhaps there is a way to autoconfig a ssh server on android from the app itself as well)

**Comment by IvanVolosyuk** [comment]
This is for really advanced users. One user sorted this all out himself and
asked me to add an option to use only local connections. I just made a
custom version of WiFi Keyboard for him, because normal user will never do
that. Https version would be secure, except it doesn't work with IP
addresses without host names.

**Comment by sudomain** [comment]
> I just made a custom version of WiFi Keyboard for him, because normal user will never do that. 

Do you still have this version? I'm very interested in being able to bind to localhost.

As an aside, [Termux](https://wiki.termux.com/wiki/Remote_Access) has two SSH implementations (dropbear and openssh) and an openssl-tool package. People that want to use password based auth could use ssh with a `sshd_config` file containing `PasswordAuthentication yes`, while people that want keys/certs could follow one of the countless tutorials available online.
